src/DynamoApplications/StartupUtils.cs

```diff
@@ -174,21 +179,85 @@ private static IUpdateManager InitializeUpdateManager()
             return um;
         }
 
+        /// <summary>
+        /// Extracts version of ASM dlls from a path by scanning for ASM dlls in the path.
+        /// Throws if ASM binaries cannot be found in the path.
+        /// </summary>
+        /// <param name="asmPath">path to directory containing asm dlls</param>
+        /// <returns></returns>
+        /// <param name="searchPattern">optional - to be used for testing</param>
+        /// <returns></returns>
+        internal static Version GetVersionFromASMPath(string asmPath, string searchPattern = "ASMAHL*.dll")
```

Comment:
I prefer this function moved to `ShapeManager` if you guys prefer this function to be ASM binaries specific.. Otherwise, like what you have done in the test, this can be a generic get file version function which we use in some other places.

Reply:
yeah, let me try to move it there. I just want to keep it internal.